### DISCUSSION TITLE
filter on foreign tables not working on embedded anymore

### DISCUSSION THREAD

**Discussion by topanion** [supabase/supabase#15759] [Questions] [answered]
Hi ! I have an issue with this request that is not working anymore. The project was working fine about 1 month and a half ago and I didn't touch the code ever since.

```js
const { data, error } = await supabase
    .from("user_friend")
    .select("*, user1(*), user2(*)")
    .or(
      "and(user1.eq." +
        user.id +
        ",status.eq.approved),and(user2.eq." +
        user.id +
        ",status.eq.approved)"
    );
```
This is part of a function fetching all "user_friend" (which is a link between two users -their ids are stored in user1 and in user2- ) where the current user.id matches user1 or user2, and where the status is approved.

It used to work just fine, as "user1" and "user2" would still be compared with eq as ids, but now it gets the error saying that embedded resources can only use is null or not is null operator. I do still need to fetch all the values from user1 and user2 for later purposes, but I didn't want to use 2 requests to the database for that.

Did something change with PostgREST ? Also, is there a way to have the same result with one single fetch ? 

Please help me

EDIT : I've found several topics using "!inner" but it seems to be only working with single comparison operator, it works with "in" or "eq" easily. My issue is to make "or" understand that I want him to compare user1.id and user2.id.

**Comment by steve-chavez** [comment]
> It used to work just fine, as "user1" and "user2" would still be compared with eq as ids, but now it gets the error saying that embedded resources can only use is null or not is null operator.

Can you share the exact error message?

**Reply by topanion** [reply]
Sure, here's what I get by using console.log on error :

![image](https://github.com/supabase/supabase/assets/127961853/8277b684-091d-459d-95ea-de219c2a83e7)

**Reply by steve-chavez** [reply]
Ok, good. Can you share the `user_friend` table definition? Seems it has FKs to other two tables so those would help too.

**Reply by topanion** [reply]
Here's user_friend : 

```sql
create table
  public.user_friend (
    id bigint generated by default as identity not null,
   created_at timestamp with time zone null default now(),
    user1 uuid null,
    user2 uuid null,
    status text null default 'pending'::text,
    constraint user_friend_pkey primary key (id),
    constraint user_friend_user1_fkey foreign key (user1) references profiles (id) on delete cascade,
    constraint user_friend_user2_fkey foreign key (user2) references profiles (id) on delete cascade
  ) tablespace pg_default;
```


And here is "profiles" :

```sql
create table
  public.profiles (
    id uuid not null,
    updated_at timestamp with time zone null,
    username text null,
    full_name text null,
    avatar_url text null,
    website text null,
    email text null,
    header_url text null,
    bio text null,
    private boolean null default false,
    constraint profiles_pkey primary key (id),
    constraint profiles_username_key unique (username),
    constraint profiles_id_fkey foreign key (id) references auth.users (id) on delete cascade,
    constraint username_length check ((char_length(username) >= 3))
  ) tablespace pg_default;
```

**Reply by steve-chavez** [reply, marked as answer]
@topanion I'm able to reproduce the error. This is a conflict with a feature introduced on PostgREST v11.0.0 (https://github.com/PostgREST/postgrest/pull/2584 - [docs](https://postgrest.org/en/latest/references/api/resource_embedding.html#null-filtering-on-embedded-resources)).

To solve it you change your query to:

```js
const { data, error } = await supabase
    .from("user_friend")
    .select("*, profile1:profiles!user1(*), profile2:profiles!user2(*)")
    .or(
      "and(user1.eq." +
        user.id +
        ",status.eq.approved),and(user2.eq." +
        user.id +
        ",status.eq.approved)"
    );
```

Let me know if that works.

---

If this affects you severely you can also request a downgrade to v10.2.0 to support@supabase.io.

**Reply by topanion** [reply]
It works perfectly, just had to do some minor tweaks :) Thank you !
